From 243b8e14a2cdd317717ad8db5bb82aae9a8d7d14 Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Mon, 2 Dec 2013 10:34:18 +0100
Subject: [PATCH] network/net-lib.sh:wait_for_if_up() wait for "state UP"

really wait for "state UP", otherwise it returns earlier than we want
---
 modules.d/40network/net-lib.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/modules.d/40network/net-lib.sh b/modules.d/40network/net-lib.sh
index 88e72ece..0aa312a0 100644
--- a/modules.d/40network/net-lib.sh
+++ b/modules.d/40network/net-lib.sh
@@ -443,7 +443,7 @@ wait_for_if_up() {
     local li
     while [ $cnt -lt 200 ]; do
         li=$(ip -o link show up dev $1)
-        [ -n "$li" ] && return 0
+        [ -n "$li" ] && [ -z "${li##*state UP*}" ] && return 0
         sleep 0.1
         cnt=$(($cnt+1))
     done
